File: backend/app/api/v1/captcha.py
from fastapi import APIRouter, Response

from app.services.captcha_service import CaptchaService

captcha_router = APIRouter(prefix="/captcha", tags=["captcha"])


@captcha_router.get("/get")
async def get_captcha():
    """
    获取验证码
    :return: 验证码图片 bytes
    """
    image_bytes, captcha_id = await CaptchaService.get_captcha()

    return Response(
        content=image_bytes,
        media_type="image/png",
        headers={"x-captcha-id": captcha_id},
    )

File: ktorBackend/src/main/kotlin/com/qianrenni/controller/captcha.kt
package com.qianrenni.controller

import com.qianrenni.services.CaptchaService
import io.ktor.server.response.*
import io.ktor.server.routing.*


fun Routing.captcha() {
    route("/captcha") {
        get("/get") {
            val captchaService = CaptchaService(call.application)
            val (captchaId, image) = captchaService.getCaptcha()
            call.response.header("x-captcha-id", captchaId)
            call.respond(image)
        }
    }
}